src/option_parser.cr

```diff
@@ -24,39 +52,70 @@ class OptionParser
     parser
   end
 
+  # Creates a new parser, with its configuration specified in the block, and uses it to parse the arguments passed to the program.
   def self.parse!
     parse(ARGV) { |parser| yield parser }
   end
 
+  # Creates a new parser.
   def initialize
     @flags = [] of String
     @handlers = [] of Handler
   end
 
+  # Creates a new parser, with its configuration specified in the block.
   def self.new
     new.tap { |parser| yield parser }
   end
 
+  # Establishes the initial message for the help printout. Typically, you want to write here the name of your program,
+  # and a one-line template of its invocation.
+  #
+  # Example:
+  #
+  #     parser.banner = "Usage: crystal [command] [switches] [program file] [--] [arguments]"
+  #
   setter banner
 
+  # Establishes a handler for a flag.
+  #
+  # Flags can (but don't have to) start with a dash. They can also have an optional argument, which will get passed to
+  # the block. Each flag has a description, which will be used for the help message.
+  #
+  # Examples of valid flags:
+  #
+  # * `-a`, `-B`
+  # * `--something-longer`
+  # * `command`
```

Comment:
I think this is a side-effect of the current implementation, I didn't know this was possible :blush:

So this is probably a bug :-)